pymc3/step_methods/hmc/trajectory.py

```diff
@@ -273,15 +283,47 @@ def _theano_single_leapfrog(H, q, p, q_grad, **theano_kwargs):
     q_new_grad = H.dlogp(q_new)
     p_new += 0.5 * epsilon * q_new_grad  # half momentum update
     energy_new = energy(H, q_new, p_new)
+    v_new = H.pot.velocity(p_new)
 
     f = theano.function(inputs=[q, p, q_grad, epsilon],
-                        outputs=[q_new, p_new, q_new_grad, energy_new], **theano_kwargs)
+                        outputs=[q_new, p_new, v_new, q_new_grad, energy_new],
+                        **theano_kwargs)
+    f.trust_input = True
+    return f
+
+
+def _theano_single_leapfrog3(H, q, p, q_grad, **theano_kwargs):
```

Comment:
That was in there by mistake. I just used this to compare leapfrog with three-stage. I took it out.